core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -427,6 +431,20 @@ private RecordCursorFactory createSpliceJoin(
                     vafList.add(new SumTimestampVectorAggregateFunction(ast.rhs.position, columnIndex));
                     continue;
                 }
+            } else if (isSingleColumnFunction(ast, "ksum")) {
+                final int columnIndex = metadata.getColumnIndex(ast.rhs.token);
+                final int type = metadata.getColumnType(columnIndex);
+                if (type == ColumnType.DOUBLE) {
+                    vafList.add(new KSumDoubleVectorAggregateFunction(ast.rhs.position, columnIndex, workerCount));
+                    continue;
+                }
+            } else if (isSingleColumnFunction(ast, "nsum")) {
+                final int columnIndex = metadata.getColumnIndex(ast.rhs.token);
+                final int type = metadata.getColumnType(columnIndex);
+                if (type == ColumnType.DOUBLE) {
+                    vafList.add(new NSumDoubleVectorAggregateFunction(ast.rhs.position, columnIndex, workerCount));
+                    continue;
+                }
             } else if (isSingleColumnFunction(ast, "avg")) {
```

Comment:
I've been thinking that avg() should have the option of using ksum and nsum.

Reply:
Yep, i agree. This PR is quite large as it is. I feel it would make sense to add avg() impls later.